src/List/NonEmpty/Zipper.elm

```diff
@@ -1204,3 +1204,37 @@ duplicateList =
 genericMove : (a -> Maybe a) -> (a -> Maybe a) -> a -> Zipper a
 genericMove f g z =
     Zipper (maybeIter f [] z) z (maybeIter g [] z)
+
+
+goToIndex : Int -> Zipper a -> Maybe (Zipper a)
+goToIndex index zipper =
+    Just (start zipper)
+        |> nTimes index next
+
+
+nTimes : Int -> (a -> Maybe a) -> Maybe a -> Maybe a
+nTimes n fn value =
+    if n <= 0 then
+        value
+
+    else
+        nTimes (n - 1) fn (Maybe.andThen fn value)
+
+
+{-| Map only the element in the zipper at the given index.
+
+    custom ['A', 'B'] 'C' ['D', 'E', 'F']
+      |> mapAtIndex 1 Char.toLower
+      --> Just <| custom ['A', 'b'] 'C' ['D', 'E', 'F']
+
+-}
+mapAtIndex : Int -> (a -> a) -> Zipper a -> Maybe (Zipper a)
```

Comment:
This is not map since the function is `a -> a`. It's not mapping one type to the other. This function just updates value with value of a same type. I would rename it to `updateAtIndex`.